src/server/api/contactsApi.js

```diff
@@ -0,0 +1,165 @@
+import { r } from '../models'
+import _ from 'lodash'
+import { getTimezoneByZip, getOptOutSubQuery } from '../../workers/jobs'
+
+import { getValidatedData } from '../../lib'
+
+function campaignaStatusShortCircuit(campaign, res) {
+  let message = ''
+  if (campaign.is_archivedj) {
+    message = 'Campaign is archived'
+  } else if (campaign.is_started) {
+    message = 'Campaign is started'
+  }
+
+  if (message) {
+    res.writeHead(403)
+    res.end(message)
+    return true
+  }
+
+  return false
+}
+
+// TODO(lperson) enforce that this is not reentrant
+// TODO(lperson) enforce auth
+export default async function contactsApi(req, res) {
+  const orgId = req.params.orgId
+  const campaignId = req.params.campaignId
+
+  if (!['GET', 'DELETE', 'POST'].includes(req.method)) {
+    res.writeHead(405, { Allow: 'GET, POST, DELETE' })
+    res.end('Not allowed')
+    return
+  }
+
+  const campaignArray = await r
+    .knex('campaign')
+    .select()
+    .where({ organization_id: orgId, id: campaignId })
+
+  if (campaignArray.length !== 1) {
+    res.writeHead(404)
+    res.end('Not found')
+    return
+  }
+
+  const campaign = campaignArray[0]
+
+  let resp = {}
+
+  switch (req.method) {
```

Comment:
Can we change this to if....else if statements (or separate methods)?

switch isn't understood by all devs these days, and as an example of an accidental removal/skip of 'break' from the GET would lead directly into the DELETE code.

Reply:
👍